app.py

```diff
@@ -31,46 +30,176 @@ def init_app(self):
         }
         
         self.engine.register_command("show_in_filesystem", self.show_in_filesystem, p)
-            
+
     def launch(self, path):
-        self.log_debug("Launching file system viewer for folder %s" % path)        
-        
-        # get the setting        
+        """
+        Given a path this method will choose the appropriate way to open that as a folder in the OS's default
+        file manager. Folders will be opened as is
+        :param path:
+        :return: None
+        """
+
+        if os.path.isfile(path):
+            self._launch_filemanager_for_file(path)
+        elif os.path.isdir(path):
+            self._launch_filemanager_for_folder(path)
+        else:
+            # possibly we have an image sequence and therefore its a symbolic path, instead check to see if
+            # the parent folder of the path is valid and try opening that.
+            # todo: The issue with this logic is that possibly it was a directory that just didn't exist,
+            # so we would just be gathering the next directory up,
+            # ideally need a better way to handle sequences.
+            parent_dir = os.path.dirname(path)
+            if parent_dir and os.path.isdir(parent_dir):
+                self._launch_filemanager_for_folder(path)
+
+            
+    def _launch_filemanager_for_folder(self, path):
+        """
+        This method will take a path to a folder and open it in the OS's default file manager.
+
+        :param path: A folder path
+        :return: None
+        """
+        self.log_debug("Launching file system viewer for folder %s" % path)
+
+        # Check that we don't have a file path.
+        if os.path.isfile(path):
+            raise ValueError(
+                "The path should be a folder not be a file path. "
+                "If you want to open a folder containing a file use _launch_filemanager_for_file. Path: %s" % path)
```

Comment:
we should not include private methods in the error message. Just say 'path 'xxxx' is not a valid directory' is sufficient. I think the logic should be `if not os.path.isdir(path):` to ensure we check other things than just files too. Does that make sense?